### PR TITLE
first step of removal of IntegralDomain

### PR DESCRIPTION
**Pull request by fchapoton**
another good step towards the removal of the auld category `IntegralDomain`

It seems reasonable to do this by chunks.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/categories/integral_domains.py [hunk 1]
```diff
@@ -143,6 +143,38 @@ def is_integral_domain(self, proof=True):
...
+
+                sage: ZZ['x'].is_field()
+                False
+            """
+            if self.is_finite():
+                return True
+            if proof:
+                raise NotImplementedError("unable to determine whether or not is a field.")
+            else:
+                return False
```
Comment (mantepse) [inline src/sage/categories/integral_domains.py]:
```suggestion
            if proof:
                raise NotImplementedError(f"unable to determine whether or not {self} is a field.")
            return False
```
I think a doctest would be good here.

**Comment by user202729** [comment]
Actually why are we doing this? Where is this discussed?

**Comment by fchapoton** [comment]
There was an historical hierarchy of such classes. It was supposed to be replaced by the `Parent` class together with the category mechanism. Somehow people doing that stopped or moved to other interests, leaving the job half done (ten years ago or more,  maybe). So we have a legacy to clean, and this is long overdue. These classes have been informally deprecated a long time ago. We will deprecate (most of) them and remove them after one year.

**Review by mantepse — APPROVED** [review]
you make me happy!

**Comment by mantepse** [comment]
OT: have you seen my question on #37158 ?

**Pull request merged**